core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -32,6 +32,17 @@ private CatalogProperties() {}
   public static final String VIEW_DEFAULT_PREFIX = "view-default.";
   public static final String METRICS_REPORTER_IMPL = "metrics-reporter-impl";
 
+  /** http proxy configuration for rest catalog */
+  public static final String PROXY_HOSTNAME = "proxy.hostname";
```

Comment:
I don't think these properties should live in this class. Since they are specific to the HTTP client, I'd suggest to move them to `HTTPClient` and prefix them with `rest.client.proxy`